### ISSUE TITLE
Windows 11 memory allocation issue with FFmpeg

### ISSUE THREAD

**Issue by beeradmoore** [microsoft/WSL#8577]
### Version

Microsoft Windows [Version 10.0.22621.160]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.102.1

### Distro Version

Ubuntu 22.09

### Other Software

Bug report contains information about FFmpeg running on Windows and WSL.

<details>
  <summary>Windows - ffmpeg 5.0</summary>

```
ffmpeg -version
ffmpeg version 5.0-essentials_build-www.gyan.dev Copyright (c) 2000-2022 the FFmpeg developers
built with gcc 11.2.0 (Rev5, Built by MSYS2 project)
configuration: --enable-gpl --enable-version3 --enable-static --disable-w32threads --disable-autodetect --enable-fontconfig --enable-iconv --enable-gnutls --enable-libxml2 --enable-gmp --enable-lzma --enable-zlib --enable-libsrt --enable-libssh --enable-libzmq --enable-avisynth --enable-sdl2 --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-libaom --enable-libopenjpeg --enable-libvpx --enable-libass --enable-libfreetype --enable-libfribidi --enable-libvidstab --enable-libvmaf --enable-libzimg --enable-amf --enable-cuda-llvm --enable-cuvid --enable-ffnvcodec --enable-nvdec --enable-nvenc --enable-d3d11va --enable-dxva2 --enable-libmfx --enable-libgme --enable-libopenmpt --enable-libopencore-amrwb --enable-libmp3lame --enable-libtheora --enable-libvo-amrwbenc --enable-libgsm --enable-libopencore-amrnb --enable-libopus --enable-libspeex --enable-libvorbis --enable-librubberband
libavutil      57. 17.100 / 57. 17.100
libavcodec     59. 18.100 / 59. 18.100
libavformat    59. 16.100 / 59. 16.100
libavdevice    59.  4.100 / 59.  4.100
libavfilter     8. 24.100 /  8. 24.100
libswscale      6.  4.100 /  6.  4.100
libswresample   4.  3.100 /  4.  3.100
libpostproc    56.  3.100 / 56.  3.100
```
</details>

<details>
<summary>Ubuntu - ffmpeg 4.4.2</summary>

```
ffmpeg version 4.4.2-0ubuntu0.22.04.1 Copyright (c) 2000-2021 the FFmpeg developers
built with gcc 11 (Ubuntu 11.2.0-19ubuntu1)
configuration: --prefix=/usr --extra-version=0ubuntu0.22.04.1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --enable-pocketsphinx --enable-librsvg --enable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
libavutil      56. 70.100 / 56. 70.100
libavcodec     58.134.100 / 58.134.100
libavformat    58. 76.100 / 58. 76.100
libavdevice    58. 13.100 / 58. 13.100
libavfilter     7.110.100 /  7.110.100
libswscale      5.  9.100 /  5.  9.100
libswresample   3.  9.100 /  3.  9.100
libpostproc    55.  9.100 / 55.  9.100
```
</details>

Also tested on the latest (and recommended) FFmpeg version
<details>
<summary>Ubuntu - ffmpeg master</summary>

```
ffmpeg version N-62110-g4d45f5acbd-static https://johnvansickle.com/ffmpeg/  Copyright (c) 2000-2022 the FFmpeg developers
built with gcc 8 (Debian 8.3.0-6)
configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc --enable-fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-libgme --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libdav1d --enable-libxvid --enable-libzvbi --enable-libzimg
libavutil      57. 26.100 / 57. 26.100
libavcodec     59. 33.100 / 59. 33.100
libavformat    59. 24.100 / 59. 24.100
libavdevice    59.  6.100 / 59.  6.100
libavfilter     8. 40.100 /  8. 40.100
libswscale      6.  6.100 /  6.  6.100
libswresample   4.  6.100 /  4.  6.100
libpostproc    56.  5.100 / 56.  5.100
```
</details>

### Repro Steps

Download the attached sample and run on Windows 10 with the `run.bat` script.
Run the same sample on Windows 10 in WSL with `run.sh`
Download the attached sample and run on Windows 11 with the `run.bat` script.
Run the same sample on Windows 11 in WSL with `run.sh`

[memory_allocate_issue.zip](https://github.com/microsoft/WSL/files/9042857/memory_allocate_issue.zip)

### Expected Behavior

Both systems produce identical videos regardless if its in Windows or WSL.

### Actual Behavior

Windows 10: Creates a sample video which has a blue box fade in and out in the upper righthand corner.
Windows 11: Creates a sample video which has a blue box fade in and out in the upper righthand corner.
Windows 10 with WSL: Creates a sample video which has a blue box fade in and out in the upper righthand corner.
Windows 11 with WSL: Fails to ever create video, repeats multiple errors of 
```
Failed to reallocate parser buffer to -2147481504
```
which I believe is generated from [this line](https://github.com/FFmpeg/FFmpeg/blob/f4098bbc3b10926f618cf89e24780c9e6ae9b8b5/libavcodec/parser.c#L229) in ffmpeg source.

### Diagnostic Logs

The reason I believe this is a WSL issue instead of a FFmpeg issue is that this it changes depending on if you are in Windows 10 or 11. We initially discovered this issue when using FFmpeg in docker (amazonlinux2) on Windows 11, it works fine in docker (amazonlinux2) on Windows 10. 

What made us even more convinced it is a WSL issue is when we turn off WSL for docker and use Hyper-V backed docker then the sample runs fine on Windows 11 docker (amazonlinux2).

Something of else I noticed is that I see RAM slowly build up to 2.04GiB before it spits the error and memory drops back down to 54MiB only to repeat this up and down pattern.

**Comment by owenschupp** [comment]
@beeradmoore could you provide the results of running `strace` and `ltrace`? Thanks!

**Comment by beeradmoore** [comment]
Something of else to note (adding to the initial post for additional info), is I see RAM slowly build up to 2.04GiB before it spits the error and memory back down to 54MiB and will repeat this pattern.

@owenschupp I ran `strace -o strace.log ./run.sh` and ` ltrace -o ltrace.log sh ./run.sh` until I saw the `Failed to reallocate parser buffer to -2147481504` message appear and then killed the process (alt+z I think). Unsure if the full results are in the log as expected on account of having to terminate them.

[strace.log](https://github.com/microsoft/WSL/files/9051138/strace.log)

ltrace was 0kb so I re-did them with out the shell script
```ltrace -o ltrace.log ffmpeg \
    -y \
    -i base_video.mp4 \
    -loop 1 \
    -i image_overlay.png \
    -filter_complex "[1:v] scale=320:118, fade=t=in:st=3:d=0.5:alpha=1, fade=t=out:st=5:d=0.5:alpha=1 [imageFade]; [0:v][imageFade] overlay=0:0:enable='between(t,3,5.5)'" \
    -t 10.000 \
    -r 30 \
    -c:v libx264 \
    -pix_fmt yuv420p \
    -an \
    render_sh.mp4
```

and the log was just:
> +++ killed by SIGKILL +++
